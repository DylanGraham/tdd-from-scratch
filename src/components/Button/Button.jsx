import * as React from "react"

const Button = () => {
  return <button className="button-primary"></button>
}

export default Button
