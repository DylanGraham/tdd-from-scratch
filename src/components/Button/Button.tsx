import * as React from "react"

interface ButtonProps {
  color?: "primary" | "secondary"
  onClick?: (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void
}

const Button: React.FunctionComponent<ButtonProps> = ({
  children,
  onClick,
  color = "primary"
}) => {
  return (
    <button
      data-testid="button"
      type="button"
      className={`button button-${color}`}
      onClick={onClick}
    >
      {children}
    </button>
  )
}

export { Button }
