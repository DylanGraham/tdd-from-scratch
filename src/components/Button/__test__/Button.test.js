import React from "react"
import { render } from "@testing-library/react"
import "@testing-library/jest-dom/extend-expect"
import Button from "../Button"

it("should render as default button", () => {
  const { container } = render(<Button />)
  expect(container.firstChild).toMatchSnapshot()
})
