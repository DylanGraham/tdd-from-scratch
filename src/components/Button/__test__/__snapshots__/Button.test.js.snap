// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render as default button 1`] = `<button />`;
